src/main/java/cmu/pasta/mu2/diff/plugin/DiffGoal.java

```diff
@@ -188,6 +203,77 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             throw new MojoExecutionException("Invalid Mutation OptLevel!");
         }
 
+        // Set filters to default value of empty string
+        if (filters == null){
+            filters = "";
+        }
+
+        // Initialize empty list of filters
+        List<MutantFilter> filterList = new ArrayList<>();
+        int i = 0;
+        char[] filterChars = filters.toCharArray();
+        String current = "";
+        Filter f;
+        int len = filterChars.length;
+
+        // Parse input string (of format "filter1:arg1,arg2,...,filter2:arg1,arg2,...") to create list of filters
+        while(i < len){
+            char c = filterChars[i];
+            if (c == ':') {
+                try {
+                    f = Filter.valueOf(current.toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    throw new MojoExecutionException("Invalid Mutation Filter!");
+                }
+                if (f == Filter.KRANDOM){
+                    
+                    i++;
+                    String arg = "";
+                    while (i < len && filterChars[i] != ','){
+                        arg += filterChars[i];
+                        i ++;
+                    }
+                    try {
+                        filterList.add(new KRandomFilter(Integer.parseInt(arg)));
+                    } catch (NumberFormatException e) {
+                        throw new MojoExecutionException("Invalid Mutation Filter Argument(s)!");
```

Comment:
Can you make these `IllegalArgumentException`s? There's three of them.

Reply:
Wouldn't the exception be a NumberFormatException since as long as the argument can be parsed into an integer the constructor can be called? I might be misunderstanding how exceptions work.

Reply:
`NumberFormatException` is appropriate for a public API whose job is explicitly to parse numbers (e.g. `Integer.parseInt`).  When the API here is "run a fuzzing session", a good error to report would be "arguments are incorrect".

However, I see from the rest of the file that `MojoExecutionException` is thrown for all user errors, presumably because it gets caught and handled by Maven for clean error reporting. So keep this code as is for now.

Reply:
I see, that makes sense - I thought you were saying we should catch an IllegalArgumentException, not throw one.